File: app/src/main/java/deakin/gopher/guardian/adapter/MedicalDiagnosticsViewPagerAdapter.java
package deakin.gopher.guardian.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.CurrentMedicalDiagnosticsFragment;
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.MedicalDiagnosticsFragment;
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.PastMedicalDiagnosticsFragment;

public class MedicalDiagnosticsViewPagerAdapter extends FragmentStateAdapter {
  private final MedicalDiagnosticsFragment parentFragment;
  private final String patient_id;

  public MedicalDiagnosticsViewPagerAdapter(
      final String patient_id, @NonNull final MedicalDiagnosticsFragment fragmentActivity) {
    super(fragmentActivity);
    parentFragment = fragmentActivity;
    this.patient_id = patient_id;
  }

  @NonNull
  @Override
  public Fragment createFragment(final int position) {
    if (0 == position) {
      parentFragment.currentFragment = new CurrentMedicalDiagnosticsFragment(patient_id);
      return parentFragment.currentFragment;
    }
    parentFragment.pastFragment = new PastMedicalDiagnosticsFragment(patient_id);
    return parentFragment.pastFragment;
  }

  @Override
  public int getItemCount() {
    return 2;
  }
}

File: app/src/main/java/deakin/gopher/guardian/adapter/MedicalDiagnosticsViewPagerAdapter.kt
package deakin.gopher.guardian.adapter

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.CurrentMedicalDiagnosticsFragment
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.MedicalDiagnosticsFragment
import deakin.gopher.guardian.view.patient.patientdata.medicaldiagnostics.PastMedicalDiagnosticsFragment

class MedicalDiagnosticsViewPagerAdapter(
    private val patientId: String,
    private val parentFragment: MedicalDiagnosticsFragment,
) : FragmentStateAdapter(
        parentFragment,
    ) {
    override fun createFragment(position: Int): Fragment {
        if (position == 0) {
            val currentFragment = CurrentMedicalDiagnosticsFragment(patientId)
            parentFragment.currentFragment = currentFragment
            return currentFragment
        }
        val pastFragment = PastMedicalDiagnosticsFragment(patientId)
        parentFragment.pastFragment = pastFragment
        return pastFragment
    }

    override fun getItemCount(): Int {
        return 2
    }
}
